text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,391 @@
+
+### Behavior of the `TraceIDRatioBased` Sampler
+
+The Sampler MUST be configured with a power-of-two probability
+expressed as `2**-s` with s being an integer in the range [0, 61]
+except for the special case of zero probability.
+
+If the context is a new root, the initial `tracestate` must be created
+with randomness value `r`, as described above, in the range [0, 61].
+If the context is not a new root, output a new `tracestate` with the
+same `r` value as the parent context.
+
+When sampled, in both cases, the context's probability value `p` is
+set to the value of `s+1` in the range [1, 63].  If the sampling
+probability is zero (the special case where `s` is undefined), use
+`p=63` the specified value for zero probability.
+
+In both cases, set the sampled bit if the outgoing `p` minus one is
+less than or equal to the outgoing `r` (i.e., `p-1 <= r`).
+
+If the context is not a new root and the incoming context's `r` value
+is not set, the implementation SHOULD notify the user of an error
+condition and follow the incoming context's `sampled` flag.
+
+The span's `log_head_adjusted_count` field is set to the outgoing `p`
+unless `r` is unknown, in which case it MUST be set to zero (unknown
+probability).
+
+### Behavior of the `ParentBased` sampler
+
+The `ParentBased` sampler is modified by this proposal.  It honors
+the W3C `sampled` flag and copies the incoming `tracestate` keys to
+the child context.
+
+The span's `log_head_adjusted_count` field is set to the incoming
+value of `p` when both `p` and `r` are defined.  When `r` is not
+defined, the span's `log_head_adjusted_count` MUST be set to 0
+indicating unknown probability, because the decision cannot be made
+consistently across the trace.
+
+### Behavior of the `AlwaysOn` Sampler
+
+The `AlwaysOn` Sampler behaves the same as `TraceIDRatioBased` with
+100% sampling probability (i.e., `p=1`).
+
+### Behavior of the `AlwaysOff` Sampler
+
+The `AlwaysOff` Sampler behaves the same as `TraceIDRatioBased` with
+zero probability (i.e., `p=63`).
+
+## Worked 3-bit example
+
+The behavior of these tables can be verified by hand using a smaller
+example.  The following table shows how these equations work where
+`r`, `p`, and `s` are limited to 3 bits.
+
+Values of `p`, which have the same encoded value and interpretation as
+for the proposed `log_head_adjusted_count` field of OTEP 170, would be
+interpreted as follows:
+
+| `p` value | Adjusted count |
+| -----     | -----          |
+| 0         | Unknown        |
+| 1         | 1              |
+| 2         | 2              |
+| 3         | 4              |
+| 4         | 8              |
+| 5         | 16             |
+| 6         | 32             |
+| 7         | 0              |
+
+Note there are only 6 non-zero, non-unknown values for the adjusted
+count. Thus there are six defined values of `r` and `s`.  The
+following table shows `r` and the corresponding selection probability,
```

Comment:
Why do we care about "`r` selection probability"? What is the objective - to generate r-value or to sample the trace?

Reply:
I think it's appropriate in an OTEP to explain technical details, they don't need to go into the specification. The reason we want to see the probability of each r-value is that it helps convince ourselves that the math works out. The r-value selection probabilities are being multiplied by adjusted counts in the example that is being worked out, that is why they are here.

Above, I've explained that an SDK is not required to use independent random bits in its TraceID. r-values can be calculated in any way that generates this distribution.